### PR TITLE
[IMP] website: allow to choose a custom color for transparent header…

### PR DESCRIPTION
**Pull request by jpr-odoo**
PURPOSE
Allow any bg *and* text colors for transparent headers

SPECIFICATION
- *one* new field for the text color
- allow to choose a custom color for bg and text color

task-2327030

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/views/website_templates.xml [hunk 1]
```diff
@@ -155,16 +155,28 @@
...
             </t>
         </t>
+        <!-- background and text color applied for only transparent header -->
+        <t t-set="header_bg_color" t-value="'header_color' in main_object and main_object.header_color"/>
+        <t t-set="header_text_color" t-value="'header_text_color' in main_object and main_object.header_text_color"/>
     </xpath>
     <xpath expr="//div[@id='wrapwrap']" position="attributes">
         <attribute name="t-attf-class" add="#{'o_header_overlay' if 'header_overlay' in main_object and main_object.header_overlay else ''}" separator=" "/>
     </xpath>
+    <!-- for transparent header background color -->
```
Comment (bvr-odoo) [inline addons/website/views/website_templates.xml:169]: `<!-- for overlay header background color -->`

File: addons/website/views/website_templates.xml [hunk 2]
```diff
@@ -155,16 +155,28 @@
     <!-- Page options -->
     <xpath expr="//div[@id='wrapwrap']" position="before">
         <t groups="website.group_website_publisher">
-            <t t-foreach="['header_overlay', 'header_color', 'header_visible', 'footer_visible']" t-as="optionName">
+            <t t-foreach="['header_overlay', 'header_color', 'header_text_color', 'header_visible', 'footer_visible']" t-as="optionName">
                 <input t-if="optionName in main_object" type="hidden" class="o_page_option_data" t-att-name="optionName" t-att-value="main_object[optionName]"/>
             </t>
         </t>
+        <!-- background and text color applied for only transparent header -->
```
Comment (bvr-odoo) [inline addons/website/views/website_templates.xml:162]: `overlay header`

File: addons/website/views/website_templates.xml [hunk 3]
```diff
@@ -155,16 +155,28 @@
...
         <attribute name="t-attf-class" add="#{'o_header_overlay' if 'header_overlay' in main_object and main_object.header_overlay else ''}" separator=" "/>
     </xpath>
+    <!-- for transparent header background color -->
     <xpath expr="//header" position="attributes">
-        <attribute name="t-attf-class" add="#{main_object.header_color if 'header_color' in main_object else ''}" separator=" "/>
+        <attribute name="t-attf-class" add="#{main_object.header_color if header_bg_color and ('bg-' in main_object.header_color) else ''}" separator=" "/>
+        <!-- applied style for inline css background color -->
+        <attribute name="t-attf-style" add="#{'background-color:' + main_object.header_color + ';' if header_bg_color and ('bg-' not in main_object.header_color) else ''}" separator=" "/>
+    </xpath>
+    <!-- for transparent header text color -->
```
Comment (bvr-odoo) [inline addons/website/views/website_templates.xml:175]: `overlay header`

**Review by bvr-odoo — CHANGES_REQUESTED** [review]

File: addons/website/models/website_page.py [hunk 4]
```diff
@@ -28,6 +28,8 @@ class Page(models.Model):
     # Page options
     header_overlay = fields.Boolean()
     header_color = fields.Char()
+    # apply text color, when header is transparent
```
[the remark this answers is not in the document]
Comment (jpr-odoo) [reply addons/website/models/website_page.py]: changes have done!

File: addons/website/views/website_templates.xml (hunk 1 above)
Comment (jpr-odoo) [reply addons/website/views/website_templates.xml]: changes have done!

File: addons/website/views/website_templates.xml (hunk 2 above)
Comment (jpr-odoo) [reply addons/website/views/website_templates.xml]: changes have done!

File: addons/website/views/website_templates.xml (hunk 3 above)
Comment (jpr-odoo) [reply addons/website/views/website_templates.xml]: changes have done!

File: addons/website/static/src/js/editor/snippets.options.js [hunk 5]
```diff
@@ -1936,10 +1943,54 @@ options.registry.topMenuColor = options.Class.extend({
...
+        return new Promise(resolve => {
+            this.trigger_up('action_demand', {
+                actionName: 'get_page_option',
+                params: ['header_overlay'],
+                onSuccess: value => resolve(!!value),
+            });
+        });
+    },
+});
+
```
[the remark this answers is not in the document]
Comment (jpr-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: changes have done!

File: addons/website/static/src/js/editor/snippets.options.js [hunk 6]
```diff
@@ -2102,11 +2109,20 @@ options.registry.topMenuColor = options.Class.extend({
...
+            this.trigger_up('action_demand', {
+                actionName: 'toggle_page_option',
+                params: [{name: 'header_text_color', value: widgetValue}],
+            });
+        } else {
+            this.trigger_up('action_demand', {
+                actionName: 'toggle_page_option',
+                params: [{name: 'header_color', value: widgetValue}],
+            });
+        }
```
Comment (bvr-odoo) [inline addons/website/static/src/js/editor/snippets.options.js:2125]: use a variable for params.name to avoid to duplicate the trigger_up ;)

**Review by bvr-odoo — CHANGES_REQUESTED** [review]

File: addons/website/static/src/js/editor/snippets.options.js (hunk 6 above)
Comment (jpr-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: Done! 👍

**Review by bvr-odoo — APPROVED** [review]

**Comment by rdeodoo** [comment]
image

**Pull request closed without merging**